common/src/main/java/io/crate/types/DataTypes.java

```diff
@@ -343,14 +343,19 @@ private static boolean safeConversionPossible(DataType type1, DataType type2) {
         entry("timestamp", TIMESTAMPZ),
         entry("interval", INTERVAL));
 
-    public static DataType ofName(String name) {
-        DataType dataType = TYPES_BY_NAME_OR_ALIAS.get(name);
+    public static DataType<?> ofName(String typeName) {
+        DataType<?> dataType = safeOfName(typeName);
         if (dataType == null) {
-            throw new IllegalArgumentException("Cannot find data type: " + name);
+            throw new IllegalArgumentException("Cannot find data type: " + typeName);
         }
         return dataType;
     }
 
+    @Nullable
+    public static DataType<?> safeOfName(String typeName) {
```

Comment:
```suggestion
    public static DataType<?> ofNameOrNull(String typeName) {
```

In other places `safe` is used to indicate that an error is thrown if the name doesn't exist. E.g `indexServiceSafe`. I'd use a different name to not overload the same expression for two different purposes - that would confuse readers.